src/main/java/net/minecraftforge/mixin/field_to_method/MobEffectInstanceMixin.java

```diff
@@ -0,0 +1,39 @@
+package net.minecraftforge.mixin.field_to_method;
+
+import net.minecraft.core.Holder;
+import net.minecraft.world.effect.MobEffect;
+import net.minecraft.world.effect.MobEffectInstance;
+import org.objectweb.asm.Opcodes;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Redirect;
+
+@Mixin(MobEffectInstance.class)
+public class MobEffectInstanceMixin {
+    /* this.effect -> this.getEffect() */
+
+    @Redirect(
+        method = {
+            "<init>(Lnet/minecraft/world/effect/MobEffectInstance;)V",
+            "getParticleOptions()Lnet/minecraft/core/particles/ParticleOptions;",
```

Comment:
Is there any way that these strings can be encoded better to provide better error when things change?
`@MethodRef("<init>", args = { MobEffectInstance.class }, ret = Void.class)` or something?
Perhaps a build script task that verifies that we have all the expected targets? 
We already have the code to generated the json that feeds the existing coremod, can something be expanded to mixins?

Reply:
I think so! I will have to write a custom annotation. But with the plugin, they can be processed in `preApply()`